luatest/server_v2.lua

```diff
@@ -0,0 +1,503 @@
+function Server:start(opts)
+    checks('table', {wait_until_ready = '?boolean'})
+    getmetatable(getmetatable(self)).start(self)
+    local wait_until_ready = true
+    if opts ~= nil and opts.wait_until_ready ~= nil then
+        wait_until_ready = opts.wait_until_ready
+    end
+    if wait_until_ready then
+        self:wait_until_ready()
+    end
+end
+
+-- TODO: Add Server:restart() function.
+
+--- Stop the server.
+-- Waits until the server process is killed, unlike the 1st version of the
+-- server class.
+function Server:stop()
+    if self.process then
+        local pid = self.process.pid
+        getmetatable(getmetatable(self)).stop(self)
+        local deadline = clock.time() + WAIT_TIMEOUT
+        while true do
+            if ffi.C.kill(pid, 0) ~= 0 then
+                break
+            end
+            if clock.time() > deadline then
+                error(('Stopping of server (alias: %s, workdir: %s, pid: %d) timed out')
+                    :format(self.alias, fio.basename(self.workdir), pid))
+            end
+            fiber.sleep(WAIT_DELAY)
+        end
+    end
+end
+
+--- Clean the server's working directory.
+-- Should be invoked only for the stopped server.
+function Server:clean()
+    fio.rmtree(self.workdir)
+    self.instance_id = nil
+    self.instance_uuid = nil
+end
+
+--- Stop the server and clean its working directory.
+function Server:drop()
+    self:stop()
+    self:clean()
+end
+
+-- Wait until the given condition is `true` (anything except `false` and `nil`).
+-- Throw an error when timeout exceeds.
+local function wait_for_condition(cond_name, server, func, ...)
+    local pid = server.process.pid
+    local deadline = clock.time() + WAIT_TIMEOUT
+    while true do
+        if func(...) then
+            return
+        end
+        if clock.time() > deadline then
+            error(('Waiting for "%s" condition on server (alias: %s, workdir: %s, pid: %d) timed out')
+                :format(cond_name, server.alias, fio.basename(server.workdir), pid))
+        end
+        fiber.sleep(WAIT_DELAY)
+    end
+end
+
+--- Wait until the server is ready after the start.
+-- It is considered that the server is operable when its `_G.ready` variable
+-- becomes `true`.
+function Server:wait_until_ready()
+    return wait_for_condition('server is ready', self, function()
+        local ok, is_ready = pcall(function()
+            self:connect_net_box()
+            return self.net_box:eval('return _G.ready') == true
+        end)
+        return ok and is_ready
+    end)
+end
+
+--- Get ID of the server instance.
+--
+-- @return number
+function Server:get_instance_id()
+    -- Cache the value when found it first time.
+    if self.instance_id then
+        return self.instance_id
+    end
+    local id = self:exec(function() return box.info.id end)
+    -- But do not cache 0 - it is an anon instance, its ID might change.
+    if id ~= 0 then
+        self.instance_id = id
+    end
```

Comment:
If we assign `0` to `instance_id`, will the condition `if self.instance_id` return true or false?

Simply speaking, is `0` a truthy or falsy value in Lua?